src/session_impl.cpp

```diff
@@ -5676,7 +5676,11 @@ namespace {
 			ip.netmask = s->netmask;
 			std::strncpy(ip.name, s->device.c_str(), sizeof(ip.name) - 1);
 			ip.name[sizeof(ip.name) - 1] = '\0';
-			s->natpmp_mapper->start(ip);
+
+			error_code ignored;
+			std::string gateway = m_settings.get_str(settings_pack::nat_pmp_gateway);
+			boost::optional<address> gateway_addr = make_address(gateway, ignored);
```

Comment:
It would make sense to move the call to `make_address()` into the condition, to only make it in case the setting is set

Reply:
That gives the following error: `Incompatible operand types ('const none_t' and 'address')`. I'll make it an if-statement, because I don't see how to get `boost::none` working in a ternary.